File: pinyougou-sellergoods-interface/src/main/java/com/lsf/pinyougou/sellergoods/service/SpecificationService.java
package com.lsf.pinyougou.sellergoods.service;

import java.util.List;

import com.lsf.pinyougou.pojo.TbSpecification;
import vo.PageResult;


/**
 * 服务层接口
 */
public interface SpecificationService {

    /**
     * 返回全部列表
     *
     * @return
     */
    List<TbSpecification> findAll();


    /**
     * 无条件分查询
     *
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);


    /**
     * 多条件分页查询
     *
     * @param pageNum  当前页
     * @param pageSize 每页记录数
     * @return
     */
    PageResult findPageLimit(TbSpecification specification, int pageNum, int pageSize);


    /**
     * 添加
     */
    void add(TbSpecification specification);


    /**
     * 修改
     */
    void update(TbSpecification specification);


    /**
     * 根据 ID 获取实体
     *
     * @param id
     * @return
     */
    TbSpecification findOne(long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    void batchDelete(Long[] ids);

}

File: pinyougou-manager-web/src/main/webapp/js/service/specificationService.js
// 规格服务层
pyg.service('specificationService', function ($http) {

    // 查询所有规格数据
    this.findAll = function () {
        return $http.get('../specification/findAll.do');
    };

    // 无条件分页查询所有规格数据
    this.findPage = function (page, size) {
        return $http.get('../specification/findPage.do?page=' + page + '&size=' + size);
    };

    // 根据 ID 查询实体
    this.findOne = function (id) {
        return $http.get('../specification/findOne.do?id=' + id);
    };

    // 增加
    this.add = function (entity) {
        return $http.post('../specification/add.do', entity);
    };

    // 修改
    this.update = function (entity) {
        return $http.post('../specification/update.do', entity);
    };

    // 批量删除
    this.batchDelete = function (selectedList) {
        return $http.get('../specification/batchDelete.do?ids=' + selectedList);
    };

    // 多条件分页查询规格数据
    this.findPageLimit = function (page, size, searchEntity) {
        return $http.post('../specification/search.do?page=' + page + "&size=" + size, searchEntity);
    };
});
